sklearn/discriminant_analysis.py

```diff
@@ -534,7 +546,13 @@ def _solve_svd(self, X, y):
         # Centers are living in a space with n_classes-1 dim (maximum)
         # Use SVD to find projection in the space spanned by the
         # (n_classes) centers
-        _, S, Vt = svd(X, full_matrices=False)
+        import dask.array as da
```

Comment:
We don't want to explicitly import array libraries or have code that checks the type in general code like this. If we need to we can add helpers in `utils/_array_api.py` to deal with differences that remain after using the Array API. Though even there we can't assume that `import dask` will work, as it isn't a dependency of scikit-learn, so we need a way to determine that dask is installed before importing it or accessing it in a different way

Reply:
I left a comment about this above https://github.com/scikit-learn/scikit-learn/pull/28588#discussion_r1515308238.

Basically, the signs on the elements in the results are sometimes flipped with dask's svd, and I wasn't sure if users of scikit-learn expect this. 

The closest thing I can find was this
https://stackoverflow.com/questions/44765682/in-sklearn-decomposition-pca-why-are-components-negative

In any case, I will clean up this PR so that either (1) this happens implicitly in array-api-compat or (2) if the code block is unnecessary, I'll just delete it.

(so we won't end up with any explicit imports in the final code).

Reply:
I removed these changs.